solutions/cf-472D.mdx

```diff
@@ -0,0 +1,148 @@
+---
+id: cf-472D
+source: Codeforces
+title: Design Tutorial Inverse the Problem
+author: Weiming Zhou
+---
+
+[Official Analysis](https://codeforces.com/blog/entry/14028)
+
+**Time Complexity:** $\mathcal{O}(N^2\log N)$ with Kruskal's MST or $\mathcal{O}(N^2)$ with $N^2$ Prims.
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+//https://codeforces.com/contest/472/submission/116947272
+
+#include <bits/stdc++.h> 
+using namespace std;
+
+using ll = long long;
+
+using vi = vector<int>;
+#define pb push_back
+#define all(x) begin(x), end(x)
+#define sz(x) (int)(x).size()
+
+using pii = pair<int,int>; //note pii, not pi
+#define f first
+#define s second
+#define mp make_pair
+
+const int MX = 2e3 + 10;
+
+int dist[MX][MX], n, par[MX], sz[MX];
+vector<pii> adj[MX];
+vector<pair<int, pii>> edges;
+
+int find(int x){
+	if(x != par[x]) par[x] = find(par[x]); 
+	return par[x];
+}
+
+void Union(int a, int b){
+	a = find(a), b = find(b);
+<<<<<<< HEAD
+	if(a == b) return ;
+=======
+	if(a == b) return;
+>>>>>>> 4235be1a4f3175cebfb05c752a7af875255688a9
```

Comment:
i think something went wrong when merging... the indentation was fine before so you can just revert your latest commit?

Reply:
uh i think i killed git by being dumb. i will delete the branch and try again